examples/get-insights-data/main.py

```diff
@@ -0,0 +1,121 @@
+            )
+        ]
+    )
+    public_api_client = PublicAPIClient(args.subdomain)
+    perspectives = fetch_course_perspectives(public_api_client, args.course_id)
+    course_data = fetch_course_data(public_api_client, args.course_id)
+
+    construct_summary(public_api_client, perspectives, course_data)
+    construct_student_insights(public_api_client, perspectives, course_data)
+
+
+def fetch_course_perspectives(public_api_client, course_id):
+    response = public_api_client.request("get", f"courses/{course_id}/perspectives")
+    return response.json()["perspectives"]
+
+
+def fetch_course_data(public_api_client, course_id):
+    response = public_api_client.request("get", f"courses/{course_id}")
+    return response.json()["course"]
+
+
+def construct_summary(public_api_client, perspectives, course):
+    headers = [
+        "Course ID",
+        "Course Title",
+        "Perspective UUID",
+        "Perspective Title",
+        "Views",
+        "Time Viewed [min]",
+        "Unique Viewers",
+    ]
+    csv_data = []
+    for perspective in perspectives:
+        perspective_data = [
+            course["id"],
+            course["name"],
+            perspective["uuid"],
+            perspective["title"],
+        ]
+        summary_csv = get_csv(
+            public_api_client, f"perspectives/{perspective['uuid']}/insights/overview"
+        )
+        for row in summary_csv:
+            if "Views" in row:
+                perspective_data.append(row[1])
+            elif "Time Viewed [min]" in row:
+                perspective_data.append(row[1])
+            elif "Unique Viewers" in row and len(row) == 2:
+                perspective_data.append(row[1])
+        csv_data.append(perspective_data)
+
+    write_csv_file(
+        f"summary-{public_api_client.subdomain}-{course['id']}.csv", headers, csv_data
+    )
+
+
+def construct_student_insights(public_api_client, perspectives, course):
+    headers = [
+        "Course ID",
+        "Course Title",
+        "Perspective UUID",
+        "Perspective Title",
+        "Name",
+        "Email",
+        "Role",
+        "Completion rate",
+    ]
+    csv_data = []
+    for perspective in perspectives:
+        perspective_data = [
+            course["id"],
+            course["name"],
+            perspective["uuid"],
+            perspective["title"],
+        ]
+        users_csv = get_csv(
+            public_api_client,
+            f"perspectives/{perspective['uuid']}/insights/users",
+            parsed=True,
+        )
+        for row in users_csv:
+            perspective_data.append(row["Name"])
+            perspective_data.append(row["Email"])
+            perspective_data.append(row["Role"])
+            perspective_data.append(row["Completion rate [%]"])
+        csv_data.append(perspective_data)
+
+    write_csv_file(
+        f"student-{public_api_client.subdomain}-{course['id']}.csv", headers, csv_data
```

Comment:
Let's rename it to `students` (plural!)

And actually it's a product(y) question whether we only want to get students, in which case we should filter for that role (both for overview and users data). All every user, in which case we should rename every `student` in this script to user.

It's not clear from the ticket.

Reply:
well, since in the ticket we have role specified as a column of the output csv I think it makes sense to fetch all and let the user (i mean the client who uses the script) decide what to do with rows that do not belong to students.

I changed the code to use user instead of student

Reply:
Sounds reasonable.